ruby/BUILD.bazel

```diff
@@ -13,12 +15,83 @@ load("//:protobuf_version.bzl", "PROTOBUF_RUBY_VERSION")
 # Ruby Runtime
 ################################################################################
 
+string_flag(
+    name = "ffi",
+    build_setting_default = "disabled",
+    values = [
+        "enabled",
+        "disabled",
+    ],
+)
+
+config_setting(
+    name = "ffi_enabled",
+    flag_values = {
+        ":ffi": "enabled",
+    },
+)
+
+config_setting(
+    name = "ffi_disabled",
+    flag_values = {
+        ":ffi": "disabled",
+    },
+)
+
+selects.config_setting_group(
+    name = "jruby_ffi",
+    match_all = [
+        ":ffi_enabled",
+        "@rules_ruby//ruby/runtime:config_jruby",
+    ],
+)
+
+selects.config_setting_group(
+    name = "jruby_native",
+    match_all = [
+        ":ffi_disabled",
+        "@rules_ruby//ruby/runtime:config_jruby",
+    ],
+)
+
+selects.config_setting_group(
+    name = "ruby_ffi",
+    match_all = [
+        ":ffi_enabled",
+        "@rules_ruby//ruby/runtime:config_ruby",
+    ],
+)
+
+selects.config_setting_group(
+    name = "ruby_native",
+    match_all = [
+        ":ffi_disabled",
+        "@rules_ruby//ruby/runtime:config_ruby",
+    ],
+)
+
+selects.config_setting_group(
+    name = "macos_ffi_enabled",
+    match_all = [
+        ":ffi_enabled",
+        "@platforms//os:osx",
+    ],
+)
+
+selects.config_setting_group(
+    name = "linux_ffi_enabled",
```

Comment:
What about windows?

Reply:
Good question. We don't have a Windows test configuration for CI today. Are there plans to introduce one?

Reply:
Not that I'm aware of, our Windows support is pretty limited.

Reply:
I'm happy to revisit this once we determine a Ruby + Windows test plan. Right now I'd have no way to confirm it works.